### PR TITLE
Verify the cmake files are properly generated

### PR DESCRIPTION
**Pull request by pzread**
Add linter to verify all generated cmake files are properly re-generated for the changes.

Sample failure run: https://github.com/iree-org/iree/actions/runs/3587710022/jobs/6038348338

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -136,3 +136,22 @@ jobs:
...
+            --output_dir ./tests/e2e/test_artifacts
+      - name: Generates E2E model test cmake files
+        run: |
+          ./build_tools/testing/generate_cmake_e2e_model_tests.py \
+            --output ./tests/e2e/models/generated_e2e_model_tests.cmake
+      - name: Checking Diff
+        run: |
+          # Make sure to pick up new files that have been added.
+          git add -A
+          git diff HEAD --exit-code
```
Comment (ScottTodd) [inline .github/workflows/lint.yml:157]: Can you share a sample run where this fails? (both to check that it actually works and to see if the output is easy to understand)

Comment (ScottTodd) [reply .github/workflows/lint.yml:157]: Oh, I see the commit history - maybe add that to the PR description too

https://github.com/iree-org/iree/actions/runs/3587710022/jobs/6038348338

Comment (pzread) [reply .github/workflows/lint.yml:157]: Added to the description: https://github.com/iree-org/iree/actions/runs/3587710022/jobs/6038348338

Comment (ScottTodd) [reply .github/workflows/lint.yml:157]: and maybe include the command to run that fixes the lint error?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -136,3 +136,22 @@ jobs:
         uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
       - name: Running check_path_lengths
         run: ./build_tools/scripts/check_path_lengths.py
+
+  generated_cmake_files:
```
Comment (GMNGeoffrey) [inline .github/workflows/lint.yml:140]: Can you add this also to https://github.com/iree-org/iree/blob/main/build_tools/scripts/lint.sh

**Review by GMNGeoffrey — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 2 above)
Comment (ScottTodd) [reply .github/workflows/lint.yml:140]: nice auto-merge :joy:

**Comment by GMNGeoffrey** [comment]
dammit, I even thought I checked that automerge wasn't on before approving :cry:

Separately, I'm thinking we should probably just create a single lint job that runs all of these, even if it does it in separate steps. Accessing a separate VM and doing a separate checkout for each of these seems a bit silly. We need a way to reliably reset the repo, but I think that should be doable.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (pzread) [reply .github/workflows/lint.yml:157]: Addressed in the follow-up #11381

**Pull request merged**